bitcoin/src/crypto/sighash.rs

```diff
@@ -56,17 +56,13 @@ impl_message_from_hash!(LegacySighash);
 impl_message_from_hash!(SegwitV0Sighash);
 
 sha256t_hash_newtype! {
-    pub struct TapSighashTag = hash_str("TapSighash");
-
-    /// Taproot-tagged hash with tag \"TapSighash\".
+    /// Taproot-tagged hash with tag "TapSighash".
     ///
-    /// This hash type is used for computing taproot signature hash."
+    /// This hash type is used for computing the taproot signature hash."
     #[hash_newtype(forward)]
-    pub struct TapSighash(_);
+    pub struct TapSighash(sha256::Hash) = hash_str("TapSighash");
 }
 
-impl_message_from_hash!(TapSighash);
```

Comment:
In 2c73f7e5b004e92df2ec1d6d9b42ab327774fce6:

Why does this go away?

Reply:
That's a mistake, right there. I had a feeling something was wrong with the `Message::from_digest` instead of `Message::from` changes. And I knew you'd catch it in review. Sorry to lean on you like that, I've literally done various combinations of all these `hashes` changes like 10 times over the last two weeks.